### PR TITLE
control-service: set build job impagePullPolicity to default to IfNot…

### PR DESCRIPTION
**Pull request by tozka**
…Present

We rarely change the version (tag) of the builder job image. There's not
point of trying to pull it every time , so this is setting it to
IfNotPresent. This way we can limit the chance of hitting docker limits.

It's made configurable using
`datajobs.deployment.builder.imagePullPolicy` system property in case in
some environments it is necessary.

Testing Done: integration test passed

Signed-off-by: Antoni Ivanov <aivanov@vmware.com>

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java [hunk 1]
```diff
@@ -59,7 +58,9 @@
    @Value("${datajobs.deployment.dataJobBaseImage:python:3.9-slim}")
    private String deploymentDataJobBaseImage;
    @Value("${datajobs.deployment.builder.extraArgs:}")
-   private String extraArgs;
+   private String builderJobExtraArgs;
+   @Value("${datajobs.deployment.builder.imagePullPolicy:IfNotPresent}")
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: Shouldn't the default value be in application.properties?

**Comment by mivanov1988** [comment]
I think we need the same for the application container. What do you think?

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java (hunk 1 above)
Comment (tozka) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: I am not sure. I have always put them here I think. But perhaps it is better for the default to be only in application.properties (alternatively in both places or only here) ?
 What is the recommended way? What are the pros/cons ?

**Comment by tozka** [comment]
> I think we need the same for the application container. What do you think?

It's already configurable: https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/helm_charts/pipelines-control-service/templates/deployment.yaml#L36

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java (hunk 1 above)
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: If someone uses it elsewhere he/she won't be able to benefit from the default value. Also it is easier to see the default values by looking application.properties.

**Review by mivanov1988 — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java (hunk 1 above)
Comment (tozka) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: > If someone uses it elsewhere he/she won't be able to benefit from the default value

What do you mean by this?

Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: If I inject (`@Value("${datajobs.deployment.builder.imagePullPolicy}")`) this property in another class it wouldn't have a default value.

Comment (tozka) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java:62]: Got it. That make sense.

**Pull request merged**